polus-precompute-volume-plugin/src/utils.py

```diff
@@ -0,0 +1,338 @@
+        mesh1bounds = mesh1.bounds
+        logger.debug('** Loaded chunk #1: {} ---- {} bytes'.format(mesh_fileobj, os.path.getsize(mesh1_path)))
+
+        # if there is only one mesh, then decompose
+        if len_files == 1:
+            num_lods = math.ceil(math.log(len(mesh1.vertices),1024))
+            ngmesh.fulloctree_decomposition_mesh(mesh1, num_lods=num_lods, 
+                    segment_id=iden, directory=output_image, quantization_bits=bit_depth)
+        # else concatenate the meshes
+        else:
+            stripped_files_middle = [idy.strip('.ply').split('_')[1:] for idy in idenfiles]
+            for i in range(len_files-1):
+                mesh2_path = os.path.join(temp_dir, idenfiles[i])
+                mesh2 = trimesh.load_mesh(file_obj=mesh2_path, file_type='ply')
+                logger.debug('** Loaded chunk #{}: {} ---- {} bytes'.format(i+2, idenfiles[i], os.path.getsize(mesh2_path)))
+                transformationmatrix = [int(trans) for trans in stripped_files_middle[i]]
+                offset = [transformationmatrix[i]/chunk_size[i] for i in range(3)]
+                middle_mesh = transformationmatrix
+                translate_middle = ([0, 1, 0, middle_mesh[0] - offset[0]],
+                                    [1, 0, 0, middle_mesh[1] - offset[1]],
+                                    [0, 0, 1, middle_mesh[2] - offset[2]],
+                                    [0, 0, 0, 1])
+                mesh2.apply_transform(translate_middle)
+                mesh1 = trimesh.util.concatenate(mesh1, mesh2)
+            num_lods = math.ceil(math.log(len(mesh1.vertices),1024))
+            ngmesh.fulloctree_decomposition_mesh(mesh1, num_lods=num_lods, 
+                    segment_id=iden, directory=output_image, quantization_bits=bit_depth)
+    except Exception as e:
+        traceback.print_exc()
+
+def build_pyramid(input_image, 
+                  output_image, 
+                  imagetype, 
+                  mesh):
+
+    """
+    This function builds the pyramids for Volume Generation and Meshes (if specified)
+
+    Parameters
+    ----------
+    input_image : str
+        Where the input directory is located
+    output_image : str
+        Where the output directory is located
+    imagetype : str
+        Specifying whether we are averaging or taking the mode of the images 
+        when blurring the images for the pyramids
+    mesh : boolean
+        Whether or not meshes are generated with segmented volumes
+
+    Returns
+    -------
+    Pyramids or volumes of input data
+    """
+
+    # Getting the intial information for the info file specification required by Neuroglancer 
+    bf = BioReader(input_image, max_workers=max([cpu_count()-1,2]))
+    bfshape = bf.shape
+    datatype = np.dtype(bf.dtype)
+    logger.info("Image Shape {}".format(bfshape))
+    logger.info("Image Datatype {}".format(datatype))
+
+    # info file specifications
+    chunk_size = [256,256,256]
+    bit_depth = 16
```

Comment:
I think these things should be set as global constants so they are easier to find and change in the future if necessary.

Also, we may wan to consider different chunk sizes for images/segmentations versus meshes. The image/segmentation formats are dense, where the meshes are sparse. We have already seen the performance issues with the precompute slide, and this plugin is supposed to have a more performant volume rendering. The chunk_size they recommend in Neuroglancer is 64x64x64, and in my testing that has also been the most performant across environments. We should use that for image/segmentation types.

Is there a reason why we can't use `bit_depth = 10`? I realize this might be splitting hairs, but if it's an easy switch then I think we should switch to 10 because 16-bit precision on chunks 256x256x256 seems like overkill.